crates/bevy_render/src/render_resource/resource_macros.rs

```diff
@@ -120,4 +120,30 @@ macro_rules! render_resource_wrapper {
     };
 }
 
+#[macro_export]
+macro_rules! define_atomic_id {
+    ($atomic_id_type:ident) => {
+        #[derive(Copy, Clone, Hash, Eq, PartialEq, Debug)]
+        pub struct $atomic_id_type(u32);
+
+        impl Default for $atomic_id_type {
```

Comment:
Does it make sense to implement `Default` for these types? It's surprising that it would return a distinct value each time you call it, I would expect `T::default() == T::default()` to hold -- this isn't a strict rule or anything, but it seems like every implementation in `std` has this property.

I think defining a `new` method would make more sense.

Reply:
Yeah, that's a valid objection. I will change it back. Clippy was not happy with new without default, but I can disable the warning.

Reply:
When you silence that clippy lint, make sure you add a comment explaining why you aren't implementing default. The reasoning I gave probably won't be obvious to everyone.

Reply:
This should do it.